File: bindings/Octave/tests/relative_error.m
function err = relative_error(x, y)
     if ~isequal(size(x), size(y))
        error("Inconsistent vector sizes");
     end
     x_norm = norm(x);
     y_norm = norm(y);
     if x_norm > 0 || y_norm > 0
         diff_norm = norm(x - y);
         err = diff_norm / max(x_norm, y_norm);
     else
        err = 0;
     end
end

File: bindings/Octave/tools/relative_error.cpp
#include "relative_error.hpp"

double relative_error(const arma::vec& x, const arma::vec& y) {
  double x_norm = arma::norm(x, "inf");
  double y_norm = arma::norm(y, "inf");

  if (x_norm > 0 || y_norm > 0) {
    double diff_norm = arma::norm(x - y, "inf");
    return diff_norm / std::max(x_norm, y_norm);
  } else {
    return 0.;
  }
}